consul/resource_consul_peering.go

```diff
@@ -0,0 +1,171 @@
+
+The ` + "`cluster_peering`" + ` resource can be used to establish the peering after a peering token has been generated.
+
+~> **Cluster peering is currently in technical preview:** Functionality associated with cluster peering is subject to change. You should never use the technical preview release in secure environments or production scenarios. Features in technical preview may have performance issues, scaling issues, and limited support.
+
+The functionality described here is available only in Consul version 1.13.0 and later.
+`,
+		Create: resourceConsulPeeringCreate,
+		Read:   resourceConsulPeeringRead,
+		Delete: resourceConsulPeeringDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// In
+			"peer_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name assigned to the peer cluster. The `peer_name` is used to reference the peer cluster in service discovery queries and configuration entries such as `service-intentions`. This field must be a valid DNS hostname label.",
+			},
+			"peering_token": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Description: "The peering token fetched from the peer cluster.",
+			},
+			"datacenter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Specifies the datacenter where the peering is established. Defaults to the agent's datacenter when not specified.",
+			},
+			"token": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Specifies the ACL token to use in the request. Takes precedence over the token used by the provider.",
+			},
+			"meta": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Specifies KV metadata to associate with the peering. This parameter is not required and does not directly impact the cluster peering process.",
+
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			// Out
+			"deleted_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"peer_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"peer_ca_pems": {
+				Type:     schema.TypeList,
+				Computed: true,
+
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"peer_server_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"peer_server_addresses": {
+				Type:     schema.TypeList,
+				Computed: true,
+
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
```

Comment:
Ditto about the two additional fields that the read will output?

For the resource we should suppress the diffs for those values as they are meant as informational to the user and not something the user can set/control.